docs/dev/stream/operators/process_function.md

```diff
@@ -322,3 +322,26 @@ ctx.timerService.registerEventTimeTimer(coalescedTime)
 {% endhighlight %}
 </div>
 </div>
+
+### Fault Tolerance
+
+Timers registered within `ProcessFunction` are fault tolerant.
+
+Timers registered within `ProcessFunction` will be checkpointed by Flink. Upon restoring, timers that are checkpointed
+from the previous job will be restored on whatever new instance is responsible for that key.
```

Comment:
Add a note that timers are synchronously checkpointed (regardless of the configuration of the state backend). Hence, a large number of timers can significantly increase checkpointing time. See optimizations section for advice to reduce the number of timers.